### PR TITLE
New dataset PSM used in paper DCdetector

### PR DESCRIPTION
**Pull request by Jad-yehya**

### CODE REVIEW

File: datasets/psm.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+)
+url_xtest = (
+    "https://drive.google.com/uc?&id=1RQH7igHhm_0GAgXyVpkJk6TenDl9rd53"
+    "&export=download"
+)
+url_ytest = (
+    "https://drive.google.com/uc?&id=1SYgcRt0DH--byFbvkKTkezJKU5ZENZhw"
+    "&export=download"
+)
+
```
Comment (tomMoral) [inline datasets/psm.py:20]: Put as global variables, with upper case name.

File: datasets/psm.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    name = "PSM"
+    install_cmd = "conda"
+    requirements = ["pandas"]
+    parameters = {
+        "debug": [False],
+    }
+
+    def get_data(self):
+        # Check if the data is already here
+        if not os.path.exists("data/PSM/PSM_train.csv"):
```
Comment (tomMoral) [inline datasets/psm.py:32]: Use `get_data_path` from `benchopt.config` to get a path for PSM:
https://benchopt.github.io/user_guide/tweak_datasets.html#paths-to-data-files

Also avoid working with `os.path` but use `pathlib.Path`

File: datasets/psm.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            response = requests.get(url_ytest)
+            with open("data/PSM/PSM_test_label.csv", "wb") as f:
+                f.write(response.content)
+
+        X_train = pd.read_csv("data/PSM/PSM_train.csv")
+        X_train.fillna(X_train.mean(), inplace=True)
+        X_train = X_train.to_numpy()
+
+        X_test = pd.read_csv("data/PSM/PSM_test.csv")
+        X_test.fillna(X_test.mean(), inplace=True)
```
Comment (tomMoral) [inline datasets/psm.py:49]: Hum are you sure? ^^
Never fill `X_test` with data from the test that you should not be able to access during the eval. Use `X_train.mean()`

**Comment by Jad-yehya** [comment]
@tomMoral Fixed in PR #13, ready for review

**Pull request merged**